src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -5745,158 +5745,145 @@ def create_nw_packet_capture(cmd, client, resource_group_name, capture_name,
-            if not workspace:
-                raise CLIError('usage error (analytics not already configured): --workspace NAME_OR_ID '
-                               '[--enabled {true|false}]')
-            if traffic_analytics_enabled is None:
-                traffic_analytics_enabled = True
-            config.flow_analytics_configuration = {
-                'network_watcher_flow_analytics_configuration': {
+            if not config['flowAnalyticsConfiguration']:
+                # must create whole object
+                if not workspace:
+                    raise CLIError('usage error (analytics not already configured): --workspace NAME_OR_ID '
+                                   '[--enabled {true|false}]')
+                if traffic_analytics_enabled is None:
+                    traffic_analytics_enabled = True
+                nsg_flow_logging['flow_analytics_configuration'] = {
                     'enabled': traffic_analytics_enabled,
                     'workspace_id': workspace.properties['customerId'],
                     'workspace_region': workspace.location,
                     'workspace_resource_id': traffic_analytics_workspace,
                     'traffic_analytics_interval': traffic_analytics_interval
                 }
-            }
-        else:
-            with cmd.update_context(config.flow_analytics_configuration.network_watcher_flow_analytics_configuration) as c:
-                # update object
-                c.set_param('enabled', traffic_analytics_enabled)
+            else:
+                nsg_flow_logging['flow_analytics_configuration'] = \
+                    config['flowAnalyticsConfiguration']['networkWatcherFlowAnalyticsConfiguration']
+                nsg_flow_logging['flow_analytics_configuration']['enabled'] = traffic_analytics_enabled
                 if traffic_analytics_workspace == "":
-                    config.flow_analytics_configuration = None
+                    nsg_flow_logging['flow_analytics_configuration'] = None
                 elif workspace:
-                    c.set_param('workspace_id', workspace.properties['customerId'])
-                    c.set_param('workspace_region', workspace.location)
-                    c.set_param('workspace_resource_id', traffic_analytics_workspace)
-                    c.set_param('traffic_analytics_interval', traffic_analytics_interval)
-
-    return client.begin_set_flow_log_configuration(watcher_rg, watcher_name, config)
+                    nsg_flow_logging['flow_analytics_configuration']['workspace_id'] = workspace.properties['customerId']
+                    nsg_flow_logging['flow_analytics_configuration']['workspace_region'] = workspace.location
+                    nsg_flow_logging['flow_analytics_configuration']['workspace_resource_id'] = traffic_analytics_workspace
+                    nsg_flow_logging['flow_analytics_configuration']['traffic_analytics_interval'] = traffic_analytics_interval
```

Comment:
It seems the property names of `config['flowAnalyticsConfiguration']['networkWatcherFlowAnalyticsConfiguration']` is camel case, however the last 4 lines use snack case name